### PR TITLE
Adds searching trace when enter key is pressed

### PR DESCRIPTION
**Pull request by zeagord**

### CODE REVIEW

**Review by codefromthecrypt — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
cc @making who asked for this in the twitter issues list :D

**Review by shakuzen — DISMISSED** [review]

**Comment by tacigar** [comment]
Thank you!!

**Review by tacigar — CHANGES_REQUESTED** [review]
I think that it is necessary to handle focus well.

Currently, we can also select dropdown menu by pressing Enter key.
So, If we press Enter key when dropdown menu is opened, both menu-selecting process and fetching process run.
It is not very good for UX...

**Comment by tacigar** [comment]
For example.

```javascript
class GlobalSearch extends React.Component {
  constructor(props) {
    ...
    this.state = {
      isConditionFocused: false, // <- Add state.
    };
    ...
  }
  ...
  handleKeyPress(event) {
    const { isConditionFocused } = this.state;
    if (event.key === 'Enter' && !isConditionFocused) {
      this.handleSearchButtonClick();
    }
  }
  ...
  handleConditionFocus() {
    this.setState({ isConditionFocused: true });
  }
  handleConditionKeyBlur() {
    this.setState({ isConditionFocused: false });
  }
  handleConditionValueBlur() {
    // Delay for avoiding to fetch
    setTimeout(() => { this.setState({ isConditionFocused: false }); }, 0);
  }
  ...
  renderSearchCondition(condition, index) {
    ...
    return (
      <SearchCondition
        {...commonProps}
        onKeyFocus={this.handleConditionFocus}
        onValueFocus={this.handleConditionFocus}
        onKeyBlur={this.handleConditionKeyBlur}
        onValueBlur={this.handleConditionValueBlur}
      >
        ...
      </SearchCondition>
    )
    ...
  }
}
```

**Comment by zeagord** [comment]
> handleConditionValueBlur

Thanks @tacigar

File: zipkin-lens/src/components/GlobalSearch/index.js [hunk 1]
```diff
@@ -313,7 +343,13 @@ class GlobalSearch extends React.Component {
       );
     }
     return (
-      <SearchCondition {...commonProps}>
+      <SearchCondition
```
Comment (tacigar) [inline zipkin-lens/src/components/GlobalSearch/index.js:354]: Ah, sorry for omitting code in example.
Please set callback props for `336`-`342` like the following.

```
<SearchCondition
  {...commonProps}
  onKeyFocus={() => {
    fetchAutocompleteValues(condition.key);
    this.handleConditionFocus();
  }
  onValueFocus{() => {
    fetchAutocompleteValues(condition.key);
    this.handleConditionFocus();
  }
  onKeyBlur={this.handleConditionKeyBlur}
  onValueBlur={this.handleConditionValueBlur}
>
  ...
</SearchCondition>
```

File: zipkin-lens/src/components/GlobalSearch/index.js [hunk 2]
```diff
@@ -145,11 +156,17 @@ class GlobalSearch extends React.Component {
     });
   }
 
+  handleKeyPress(event) {
```
Comment (tacigar) [inline zipkin-lens/src/components/GlobalSearch/index.js]: Nit: How about to rename `handleKeyDown` ?

**Comment by zeagord** [comment]
Makes sense. slSince the keypress event it deprecated. Renaming it to
KeyDown will be a better choice.

**Comment by tacigar** [comment]
Thank you!!!
LGTM!!

**Review by tacigar — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
good work compadres

**Comment by making** [comment]
awesome!

**Comment by tacigar** [comment]
This PR solves one of #2351 

>  it'd be nice if pressing enter executed the search

**Pull request merged**